### PR TITLE
statistics: fix a statistics GC problem that can cause duplicated fm-sketch records

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary: statistics: fix a statistics GC problem that can cause duplicated fm-sketch records

### What is changed and how it works?

statistics: fix a statistics GC problem that can cause duplicated fm-sketch records
### Check List 

Tests 

- Unit test

### Release note 

- statistics: fix a statistics GC problem that can cause duplicated fm-sketch records

### CODE REVIEW

**Review by Reminiscent — APPROVED** [review]
LGTM

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -1004,6 +1004,9 @@ func (h *Handle) SaveStatsToStorage(tableID int64, count int64, isIndex int, hg
 		}
 	}
 	if fmSketch != nil {
+		if _, err := exec.ExecuteInternal(ctx, "delete from mysql.stats_fm_sketch where table_id = %? and is_index = %? and hist_id = %?", tableID, isIndex, hg.ID); err != nil {
+			return err
+		}
```
Comment (rebelice) [inline statistics/handle/handle.go:1009]: Maybe we should delete it when `fmSketch == nil` just like `topN`.

**Comment by heipark** [comment]
issue id: https://github.com/pingcap/tidb/issues/29968

**Pull request merged**